modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -456,7 +458,11 @@ private void onEvent(CoreEvent event, ExecutorCallback executorCallback, Context
       OperationExecutionFunction operationExecutionFunction = (parameters, operationEvent, callback) -> {
         setOperationExecutionParams(location, event, configuration, parameters, operationEvent, callback, ctx);
 
-        fluxSupplier.get().next(operationEvent);
+        if (fluxSupplier instanceof TransactionAwareFluxSinkSupplier && TransactionUtils.isTxActive(ctx)) {
```

Comment:
is this instanceof and cast necessary. Isn't there other way? Can't the fluxSupplier internally decide whehter it is in a transaction or not? I think it shouldn't be responsibility from the ComponentMessageProcessor to decide the tupo of fluxSink that it receives.